internal/internal_task_handlers.go

```diff
@@ -1819,6 +1821,26 @@ func (ath *activityTaskHandlerImpl) Execute(taskList string, t *s.PollForActivit
 
 	ctx, span := createOpenTracingActivitySpan(ctx, ath.tracer, time.Now(), activityType, t.WorkflowExecution.GetWorkflowId(), t.WorkflowExecution.GetRunId())
 	defer span.Finish()
+
+	if ath.autoHeartbeat && t.HeartbeatTimeoutSeconds != nil && *t.HeartbeatTimeoutSeconds > 0 {
+		go func() {
+			autoHbInterval := time.Duration(*t.HeartbeatTimeoutSeconds) * time.Second / 2
+			ticker := time.NewTicker(autoHbInterval)
+			defer ticker.Stop()
+			for {
+				select {
```

Comment:
I am not super familar with this client yet. Should this goroutine also listen to :
```
	closeCh               chan struct{}
	workerStopChannel     <-chan struct{}
```

Reply:
Checked. Added workerStopChannel. I don't think we need to check closeCh an it's the same as ctx.Done.